library/src/main/java/com/schibsted/spain/barista/rule/cleardata/internal/FileOperations.kt

```diff
@@ -0,0 +1,33 @@
+package com.schibsted.spain.barista.rule.cleardata.internal
+
+import android.support.test.InstrumentationRegistry.getTargetContext
+import java.io.File
+
+class FileOperations {
+
+  private val appContext = getTargetContext()
+  private val androidDirectories = arrayOf(appContext.cacheDir, appContext.filesDir)
+
+  fun deleteFile(file: File) {
+    file.delete()
+  }
+
+  fun getAllFilesRecursively(): Collection<File> {
+    return androidDirectories.flatMap { listFileTree(it) }
+  }
+
+  private fun listFileTree(dir: File?): Collection<File> {
+    if (dir?.listFiles() == null) {
+      return emptySet()
+    }
+    val fileTree = HashSet<File>()
+    dir.listFiles().forEach { entry ->
```

Comment:
This is fine but you can also do `return dir.listFiles().map{}`... and not declaring `fileTree` variable.

Reply:
Hmm but it's doing different operations depending if it's a file or a directory. `add` vs `addAll`.
How would you do it with a map?

Reply:
Maybe something like this: (quite ugly, but functional)

dir.listFiles().stream().map(entry -> {
   if (entry.isFile) {
        return Arrays.asList(entry)
      } else {
        return listFileTree(entry)
      }
}).flatMap(entry -> entry.stream())
.collect(Collectors.toList());

Reply:
I just found out an even nicer way...

```kotlin
return dir.walkTopDown()
        .filter { it.isFile }
        .toList()
```

Kotlin will never stop surprising me 😂